spec/rubocop/cop/rspec/void_expect_spec.rb

```diff
@@ -44,4 +44,62 @@
       end
     RUBY
   end
+
+  it 'ignores unrelated expect method in an it block' do
+    expect_no_offenses(<<~RUBY)
+      it 'something' do
+        MyObject.expect
+      end
+    RUBY
+  end
+
+  it 'rejects bare `expect` in an it block' do
+    expect_offense(<<~RUBY)
+      it 'something' do
+        expect
+        ^^^^^^ Do not use `expect()` without `.to` or `.not_to`. Chain the methods or remove it.
+      end
+    RUBY
+  end
+
+  context 'when expect has no parent node' do
+    it 'still registers an offense' do
+      expect_offense(<<~RUBY)
+        expect(something)
+        ^^^^^^^^^^^^^^^^^ Do not use `expect()` without `.to` or `.not_to`. Chain the methods or remove it.
+      RUBY
+    end
+
+    it 'rejects bare expect with no arguments' do
+      expect_offense(<<~RUBY)
+        expect
+        ^^^^^^ Do not use `expect()` without `.to` or `.not_to`. Chain the methods or remove it.
+      RUBY
+    end
+
+    it 'rejects bare `expect` in a block' do
+      expect_offense(<<~RUBY)
+        method { expect }
+                 ^^^^^^ Do not use `expect()` without `.to` or `.not_to`. Chain the methods or remove it.
+      RUBY
+    end
+
+    it 'accepts expect.to with no arguments' do
```

Comment:
Is this example necessary?
I think ‘to’ expects at least one argument.
Why do we need to cover what is not valid syntax?

Reply:
This isn't necessary. I was just showing that that Cop will reject `expect` on main but accept `expect.to` on main.

I can probably remove most of the variations on this with no impact on coverage.

Reply:
Do you think it is reasonable to fix the cop to check ‘inside_example?’ somewhere and remove the “no parent” check from the cop and related spec contexts altogether?

Thanks for this yak shaving! And sorry for an attempt to drag you even further 😅

Reply:
I don't think it's a good fit for the branch coverage PR. But I do think it's worth looking into!

Reply:
It looks like an easy fix to copy that over:
<img width="609" alt="Screenshot 2024-10-14 at 12 44 08 PM" src="https://github.com/user-attachments/assets/1368f236-046b-40f9-ab68-801ca61e989d">

But it breaks the branch coverage again.

Reply:
(the `on_send` copy of `inside_example?` is covered)

Reply:
Draft PR here: 
https://github.com/rubocop/rubocop-rspec/pull/1975

Reply:
This has been an interesting exercise. I'm a little bit of a branch coverage skeptic, and a huge line coverage fan. So it's interesting to see this has revealed an issue worth addressing.